enhancements/oc/image-content-source-policy-awareness.md

```diff
@@ -0,0 +1,87 @@
+---
+title: image-content-source-policy-awareness
+authors:
+  - "@sallyom"
+reviewers:
+  - "@smarterclayton"
+  - "@soltysh"
+  - "@wking"
+approvers:
+  - "@smarterclayton"
+creation-date: 2020-05-19
+last-updated: 2020-05-19
+status: provisional
+---
+
+# ImageContentSourcePolicy (ICSP) Awareness
+
+ICSP allows OpenShift (CVO, CRI-O) to check down a list of possible mirrors to find an image with the matching digest it is
+looking for.  `oc` should do the same.  
+
+There have been several bugs opened around the experience of a 
+user in a disconnected environment using `oc adm release` commands.  If
+using a mirrored image and the mirrored source registry is connected, 
+the following commands do not succeed when in a disconnected environment:
+
+```console
+$ oc adm release extract --tools aprivateregistry/repo/name:tag
+$ oc adm release mirror aprivateregistry/repo/name:tag --to someregistry/repo/name
+$ oc adm must-gather
+```
+
+This is because the mirrored image tags (the individual component images from a payload)
+retain references to the mirrored registry, usually something like 
+`quay.io/openshift-release-dev/ocp-v4.0-art-dev`.  
+
+There needs to be logic in `oc` to look for `ImageContentSourcePolicy` either from a file or from a cluster.
+`oc` should look for `ICSP` in the cluster/current context if connected, and if not, from a file in an expected location,
+probably from the current directory.  `oc` should gather information about RepositoryDigestMirrors from ICSP and use that
+when extracting or mirroring images, or when running must-gather.
+
+Current bugs regarding this Issue:   
+* https://bugzilla.redhat.com/show_bug.cgi?id=1823839
+* https://bugzilla.redhat.com/show_bug.cgi?id=1823143 and also for 4.3, 4.5, 4.6
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+* In what order should `oc` gather info about an image?  Need to order the following:
+    * ICSP from file
+    * ICSP from cluster
+    * image reference from user-given image
+
+* Should oc adm release mirror write an ICSP file to the current directory rather than a print to stdout?
```

Comment:
I see this is the case for `oc adm catalog mirror` but not `oc adm release mirror`